### DISCUSSION TITLE
Command to open DirSheet with `--dir-recurse`, but only for selected directories

### DISCUSSION THREAD

**Discussion by daviewales** [saulpw/visidata#1781] [Ideas]
I've just discovered the `--dir-recurse` option for `DirSheet`. It's very cool, because I was literally just starting to think about writing a plugin to do this, and it already does it.

But... It would be cool if you could open a directory: `vd .`
Then select just a couple of subdirectories: `ss`
Then open a `--dir-recurse DirSheet` for just those selected subdirectories.

It would also be cool to do be able to do this for zip files and Excel files.

**Comment by saulpw** [comment]
So you kind of want a `g Enter` that opens a single sheet with all the underlying entries appended.  Do you ever want only 1 level down for those, or do you always want them as deep as they go?

**Comment by daviewales** [comment]
Hmm... The motivating use case is that I have a dataset which comes as multiple separate excel files, but each excel file contains exactly the same sheets and fields. 

I want to be able to quickly push all the sheets from all the Excel files.

I start in a dir sheet, where I can select the Excel files I want. I imagine a single command to show me all the Excel sheets recursively, based on the files I've selected. Then I can immediately select the matching sheets by pattern, and concatenate them into single combined tables. Currently I have to dive into each Excel file separately to load the sheets, then go back to the sheets sheet to concatenate them together.

Once I imagined this kind of recursive flattening, I thought it could be generally useful if you have multiple directories or zip files you want to flatten as well.

My original idea was to recourse indefinitely, although I thought perhaps there could be different commands to only recurse zips, or only dirs, only Excels.

My personal use case is satisfied by single depth recursion, as Excel files are only single depth.

I also thought this could be relevant to JSON files as well, as they open by default with a single level of the tree, similar to dir, excel, zip, etc.

**Reply by saulpw** [reply]
Ah okay, this is different!  `dir_recurse` would only work for directories, whereas it sounds like you want to go deeper than that.

One thing you could do for your motivating use case, is load all the Excel files, then join their index sheets with the `concat` type.  Now you have one index with the sub-sheets from all the Excel files, and you can select any number of them and then `g Enter` to push them all as individual sheets.  Or `&` with `append` or `concat` to combine them all into one sheet.
